plugins/auth-backend/api-report.md

```diff
@@ -626,6 +752,236 @@ export type ProfileInfo = {
+                }
+              | undefined;
+          }
+        | undefined,
+    ) => AuthProviderFactory;
+    resolvers: never;
+  }>;
+  auth0: Readonly<{
+    create: (
+      options?:
+        | {
+            authHandler?: AuthHandler<OAuthResult> | undefined;
+            signIn?:
+              | {
+                  resolver: SignInResolver<OAuthResult>;
+                }
+              | undefined;
+          }
+        | undefined,
+    ) => AuthProviderFactory;
+    resolvers: never;
+  }>;
+  awsAlb: Readonly<{
+    create: (
+      options?:
+        | {
+            authHandler?: AuthHandler<AwsAlbResult> | undefined;
+            signIn: {
+              resolver: SignInResolver<AwsAlbResult>;
+            };
+          }
+        | undefined,
+    ) => AuthProviderFactory;
+    resolvers: never;
+  }>;
+  bitbucket: Readonly<{
+    create: (
+      options?:
+        | {
+            authHandler?: AuthHandler<OAuthResult> | undefined;
+            signIn?:
+              | {
+                  resolver: SignInResolver<OAuthResult>;
+                }
+              | undefined;
+          }
+        | undefined,
+    ) => AuthProviderFactory;
+    resolvers: Readonly<{
+      lookupUsernameAnnotation(): SignInResolver<OAuthResult>;
+      lookupUserIdAnnotation(): SignInResolver<OAuthResult>;
+    }>;
+  }>;
+  gcpIap: Readonly<{
+    create: (options: {
+      authHandler?: AuthHandler<GcpIapResult> | undefined;
+      signIn: {
+        resolver: SignInResolver<GcpIapResult>;
+      };
+    }) => AuthProviderFactory;
+    resolvers: never;
+  }>;
+  github: Readonly<{
+    create: (
+      options?:
+        | {
+            authHandler?: AuthHandler<GithubOAuthResult> | undefined;
+            signIn?:
+              | {
+                  resolver: SignInResolver<GithubOAuthResult>;
+                }
+              | undefined;
+            stateEncoder?: StateEncoder | undefined;
+          }
+        | undefined,
+    ) => AuthProviderFactory;
+    resolvers: Readonly<{
+      byUsername: () => SignInResolver<GithubOAuthResult>;
+    }>;
+  }>;
+  gitlab: Readonly<{
+    create: (
+      options?:
+        | {
+            authHandler?: AuthHandler<OAuthResult> | undefined;
+            signIn?:
+              | {
+                  resolver: SignInResolver<OAuthResult>;
+                }
+              | undefined;
+          }
+        | undefined,
+    ) => AuthProviderFactory;
+    resolvers: never;
+  }>;
+  google: Readonly<{
+    create: (
+      options?:
+        | {
+            authHandler?: AuthHandler<OAuthResult> | undefined;
+            signIn?:
+              | {
+                  resolver: SignInResolver<OAuthResult>;
+                }
+              | undefined;
+          }
+        | undefined,
+    ) => AuthProviderFactory;
+    resolvers: Readonly<{
+      byEmailLocalPart: () => SignInResolver<unknown>;
+      lookupEmailAnnotation(): SignInResolver<OAuthResult>;
```

Comment:
hmm some are `by`, some are `lookup` ... i wonder if the word `catalog` should be in there too? not sure 🚲 🏠

Reply:
- `emailLocalPartMatchingUserEntityName`
- `emailMatchingUserEntityAnnotation`

Reply:
Yeah happy to discuss this more. This is just a pattern I came up with after trying some stuff, did go for some more verbose options earlier on.  The `by` prefix is supposed to indicate a direct mapping to an entity ref, while `lookup` is used for cases where we need to query based on some filter. I wanted to keep them a bit short given that they're sccessed 3 layers in

Reply:
whether it requires a lookup or not may perhaps be a technicality rather than task oriented tho

Reply:
Don't mind the extra chars too much, it's not gonna be used in more than one single line of code in the user's backend :)

Reply:
Main thing is, would be good to make it clear (1) what's the source field used, (2) that it targets users in the catalog specifically, and (3) what it matches on